core/src/window_service.rs

```diff
@@ -187,15 +186,6 @@ pub(crate) fn should_retransmit_and_persist(
         } else if !verify_shred_slot(shred, root) {
             inc_new_counter_debug!("streamer-recv_window-outdated_transmission", 1);
             false
-        } else if shred.version() != shred_version {
-            inc_new_counter_debug!("streamer-recv_window-incorrect_shred_version", 1);
-            false
-        } else if shred.index() >= MAX_DATA_SHREDS_PER_SLOT as u32 {
-            inc_new_counter_warn!("streamer-recv_window-shred_index_overrun", 1);
-            false
-        } else if shred.sanitize().is_err() {
-            inc_new_counter_warn!("streamer-recv_window-invalid-shred", 1);
-            false
```

Comment:
```rust
} else if shred.version() != shred_version {
```
moved earlier in the pipeline in the shred-fetch stage.

```
} else if shred.index() >= MAX_DATA_SHREDS_PER_SLOT as u32 {
```
is already checked in `shred.sanitize()`.

```rust
} else if shred.sanitize().is_err() {
```
shreds are already sanitized when de-serialized.